### PR TITLE
chore(tests): Improves service provider test and coverage

### PR DESCRIPTION
**Pull request by mrclay**
The test now collects all registered names out of the DiContainer, makes sure each is present in the `@property-read` declarations, and now tests the return types of _all_ services (9 more).

Refactors DatabaseQueue and lib/configuration, and updates the PHPunit bootstrap to avoid using ext/mysql during instantiation.
- [ ] One more LGTM

### CODE REVIEW

File: engine/classes/Elgg/Queue/DatabaseQueue.php [hunk 1]
```diff
@@ -86,7 +93,9 @@ public function clear() {
 	 */
 	public function size() {
 		$prefix = $this->db->getTablePrefix();
-		$result = $this->db->getDataRow("SELECT COUNT(id) AS total FROM {$prefix}queue WHERE name = '$this->name'");
+		$name = $this->db->sanitizeString($this->name);
+
+		$result = $this->db->getDataRow("SELECT COUNT(id) AS total FROM {$prefix}queue WHERE name = '$name'");
```
Comment (ewinslow) [inline engine/classes/Elgg/Queue/DatabaseQueue.php]: I approve of this change, but can we put it in a separate commit?

Comment (ewinslow) [inline engine/classes/Elgg/Queue/DatabaseQueue.php]: Or was this necessary in order to get the improved test coverage?

File: engine/lib/configuration.php [hunk 2]
```diff
@@ -275,6 +275,33 @@ function _elgg_load_site_config() {
...
+ * Set up CONFIG->cookies. (this is for unit testing)
+ *
+ * @see phpunit/bootstrap.php
+ *
+ * @param stdClass $CONFIG Elgg's config object
+ * @access private
+ */
+function _elgg_configure_cookies($CONFIG) {
+	// set cookie values for session and remember me
+	if (!isset($CONFIG->cookies)) {
```
Comment (ewinslow) [inline engine/lib/configuration.php]: Instead of side effects on config, perhaps we could return the cookies array? Then the caller can be responsible for merging the result into `$CONFIG` i.e.: `$CONFIG->cookies = _elgg_configure_cookies();`

**Comment by ewinslow** [comment]
Great stuff!

File: engine/classes/Elgg/Queue/DatabaseQueue.php (hunk 1 above)
Comment (mrclay) [inline engine/classes/Elgg/Queue/DatabaseQueue.php]: Yes, it was needed to get all DB work out of the constructor so the svc could be made without warnings

File: engine/lib/configuration.php (hunk 2 above)
Comment (mrclay) [inline engine/lib/configuration.php]: Is it really necessary? Ideally this procedural setup won't live long anyway.

Comment (ewinslow) [inline engine/lib/configuration.php]: Not really, but something to keep in mind for the future. Think return values not side effects. Much more testable and typeable that way!

Comment (mrclay) [inline engine/lib/configuration.php]: Agreed. I went to refactor this and noticed it array merges the values into possibly existing arrays in $CONFIG. Reworking this is a can of worms not worth my time at the moment.

Comment (ewinslow) [inline engine/lib/configuration.php]: Yup sgtm

**Comment by ewinslow** [comment]
@Elgg/owners looking for one more person to review this.
